services/ui-src/src/App.js

```diff
@@ -7,63 +7,122 @@ import { Auth } from "aws-amplify";
 import Header from "./components/layout/Header";
 import Footer from "./components/layout/Footer";
 import config from "./config";
-import { getLocalUserInfo } from "./libs/user";
+import { currentUserInfo } from "./libs/user";
+import { getUserByUsername, createUser, updateUser } from "./libs/api";
 
-function App({ userData }) {
+function App() {
   const [isAuthenticating, setIsAuthenticating] = useState(true);
   const [isAuthenticated, userHasAuthenticated] = useState(false);
+  const [isAuthorized, setIsAuthorized] = useState(false);
+  const [user, setUser] = useState();
   const history = useHistory();
 
   useEffect(() => {
-    onLoad();
-  });
+    async function onLoad() {
```

Comment:
please lets refactor the onLoad function, you can break up out and have helper functions you call in onLoad function. for example you have a function that validates the user info, that one u'll need. this was we can also use that function to validate any object through out the project.